cmd/troubleshoot/cli/run.go

```diff
@@ -123,16 +127,72 @@ func runTroubleshoot(v *viper.Viper, arg string) error {
 		close(finishedCh)
 	}()
 
-	archivePath, err := runCollectors(v, *collector, additionalRedactors, progressChan)
+	archivePath, err := runCollectors(v, supportBundleSpec.Spec.Collectors, additionalRedactors, progressChan)
 	if err != nil {
 		return errors.Wrap(err, "run collectors")
 	}
 
 	fmt.Printf("\r%s\r", cursor.ClearEntireLine())
 
-	if len(collector.Spec.AfterCollection) == 0 {
+	// upload if needed
+	fileUploaded := false
+	if len(supportBundleSpec.Spec.AfterCollection) > 0 {
+		for _, ac := range supportBundleSpec.Spec.AfterCollection {
+			if ac.UploadResultsTo != nil {
+				if err := uploadSupportBundle(ac.UploadResultsTo, archivePath); err != nil {
+					c := color.New(color.FgHiRed)
+					c.Printf("%s\r * Failed to upload support bundle: %v\n", cursor.ClearEntireLine(), err)
+				} else {
+					fileUploaded = true
+				}
+			} else if ac.Callback != nil {
+				if err := callbackSupportBundleAPI(ac.Callback, archivePath); err != nil {
+					c := color.New(color.FgHiRed)
+					c.Printf("%s\r * Failed to notify API that support bundle has been uploaded: %v\n", cursor.ClearEntireLine(), err)
+				}
+			}
+		}
+
+	}
+
+	// perform analysis, if possible
+	if len(supportBundleSpec.Spec.Analyzers) > 0 {
+		tmpDir, err := ioutil.TempDir("", "troubleshoot")
+		if err != nil {
+			c := color.New(color.FgHiRed)
+			c.Printf("%s\r * Failed to make directory for analysis: %v\n", cursor.ClearEntireLine(), err)
+		}
+
+		f, err := os.Open(archivePath)
```

Comment:
@CodeLingoBot capture defer the closing of files
Defer the closure of files when opening them to ensure cleanup regardless of success/failure.

See exiting rule template "defer close file" in the Effective Go rule template set.